src/nvim/os/input.c

```diff
@@ -7,13 +7,27 @@
 
 #include <uv.h>
 
+// For ui.h and channel.h we want to have 'extern' global
+// variables, so we need to include them before we define
+// EXTERN to an empty value.
+#include "nvim/macros.h"
+
+#include "nvim/ui.h"
+#include "nvim/msgpack_rpc/channel.h"
+
+// We want to define global variables here, so set EXTERN
+// to an emtpy define and set INIT()
```

Comment:
typo. Though the comment doesn't really explain anything. "set EXTERN to empty" just above `#undef EXTERN` is not helpful...

Reply:
This should be done by including `os/input.h` in `main.c`, as is done exactly all other global variable definitions.

This currently changes the interpretation of 14 unrelated header file which is not an improvement in any way, shape or form. "Robustness" improvements like this should be done by following some set of rules, not by randomly changing #defines and #includes around until it complies without warnings by accident.

Reply:
Sorry, I'm new to the neovim codebase and need to find my way first. Don't blame me for doing this wrong. I did not `#define EXTERN` it was already in that file.

 I'm trying to fix this and get it working correctly. It looks like we just need to remove the `#define EXTERN` from that file then. The `nvim/os/input.h` is already correctly included in main.c

Reply:
Hmm are you sure? I can't find "#define EXTERN` anywhere except `macros.h` and `main.c` .

Reply:
Ups :-) Should be fixed now.